modules/persistent-storage-csi-azure-file-cross-sub-dynamic-provisioning-procedure.adoc

```diff
@@ -0,0 +1,131 @@
+$ mi_id=$(oc -n openshift-cluster-csi-drivers get secret azure-file-credentials -o jsonpath='{.data.azure_client_id}' | base64 --decode)
+
+$ az identity list --query "[?clientId=='${mi_id}'].{Name:name}" --output tsv
+----
+
+. Grant the Azure identity (service principal or managed identity) permission to access the resource group in another Subscription B where you want to provision the Azure File share by doing one of the following:
+
+* Run the following Azure CLI command:
++
+[source,terminal]
+----
+az role assignment create \
+  --assignee <object-id-or-app-id> \
+  --role <role-name> \
+  --scope /subscriptions/<subscription-id>/resourceGroups/<resource-group>/providers/Microsoft.Storage/storageAccounts/<storage-account-name>
+----
++
+Where:
++
+`<object-id-or-app-id>`: The service principal or managed identity that you obtained from the previous step, such as `sp_id` or `mi_id`.
++
+`<role-name>`: Role name. Contributor or your own role with required permissions.
++
+`<subscription-id>`: Cross subscription B ID.
++
+`<resource-group-name>`: Cross subscription B resource group name.
++
+Or
++
+* Log in to the Azure portal and on the left menu, click *Resource groups*:
+
+.. Choose the resource group in Subscription B to which you want to assign a role by clicking *resource group* -> *Access control (IAM)* -> *Role assignments* tab to view current assignments, and then click *Add* > *Add role assignment*.
+
+.. On the *Role* tab, choose the contributor role to assign, and then click *Next*. You can also create and choose your own role with required permission.
+
+.. On the *Members* tab: 
+... Choose an assignee by selecting the type of assignee: user, group, or service principal (or managed identity). 
+... Click *Select members*. 
+... Search for, and then select the desired service principal or managed identity recorded in the previous step. 
+... Click *Select* to confirm.
+.. On the *Review + assign* tab, review the settings.
+
+.. To finish the role assignment, click *Review + assign*.
++
+[NOTE]
+====
+If you only want to use a specific storage account to provision the Azure File share, you can also obtain the Azure identity (service principal or managed identity) permission to access the storage account by using similar steps.
+====
+
+. Create an Azure File storage class by using a similar configuration to the following:
++
+.Example Azure File storage class YAML file
+[source, yaml]
+----
+allowVolumeExpansion: true
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: <sc-name-cross-sub> <1>
+mount options:
+  - mfsymlinks
+  - cache=strict
+  - nosharesock
+  - actimeo=30
+parameters:
+  subscriptionID: <xxxx-xxxx-xxxx-xxxx-xxxx> <2>
+  resourceGroup: <resource group name> <3>
+  storageAccount: <storage account> <4> 
+  skuName: <skuName> <5>
+provisioner: file.csi.azure.com
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+----
+<1> The name of the storage class
+<2> The cross account subscription ID
```

Comment:
Same here i think we should avoid using the word cross in snippets this can be misleading.
"Call it subscription B ID" or "Subscription ID where the file share resides."

Reply:
Done.